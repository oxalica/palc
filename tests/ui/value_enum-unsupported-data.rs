use palc::{__private::ValueEnum, ValueEnum};

#[derive(ValueEnum)]
struct Unit;

#[derive(ValueEnum)]
struct Tuple();

#[derive(ValueEnum)]
struct Named {}

#[derive(ValueEnum)]
union Union {
    a: (),
}

struct AssertImpl
where
    Unit: ValueEnum,
    Tuple: ValueEnum,
    Named: ValueEnum;

fn main() {}
