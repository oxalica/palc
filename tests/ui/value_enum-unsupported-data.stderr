error: only enums are supported
 --> tests/ui/value_enum-unsupported-data.rs:3:10
  |
3 | #[derive(ValueEnum)]
  |          ^^^^^^^^^
  |
  = note: this error originates in the derive macro `ValueEnum` (in Nightly builds, run with -Z macro-backtrace for more info)

error: only enums are supported
 --> tests/ui/value_enum-unsupported-data.rs:6:10
  |
6 | #[derive(ValueEnum)]
  |          ^^^^^^^^^
  |
  = note: this error originates in the derive macro `ValueEnum` (in Nightly builds, run with -Z macro-backtrace for more info)

error: only enums are supported
 --> tests/ui/value_enum-unsupported-data.rs:9:10
  |
9 | #[derive(ValueEnum)]
  |          ^^^^^^^^^
  |
  = note: this error originates in the derive macro `ValueEnum` (in Nightly builds, run with -Z macro-backtrace for more info)

error: only enums are supported
  --> tests/ui/value_enum-unsupported-data.rs:12:10
   |
12 | #[derive(ValueEnum)]
   |          ^^^^^^^^^
   |
   = note: this error originates in the derive macro `ValueEnum` (in Nightly builds, run with -Z macro-backtrace for more info)
